src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -196,22 +221,46 @@ private void checkIfSemicolonIsInDifferentLineThanPrevious(DetailAST ast) {
             currentStatement = ast.getPreviousSibling();
         }
         if (isInLambda) {
-            int countOfSemiInCurrentLambda = countOfSemiInLambda.pop();
-            countOfSemiInCurrentLambda++;
-            countOfSemiInLambda.push(countOfSemiInCurrentLambda);
-            if (!inForHeader && countOfSemiInCurrentLambda > 1
-                    && isOnTheSameLine(currentStatement,
-                    lastStatementEnd, forStatementEnd,
-                    lambdaStatementEnd)) {
-                log(ast, MSG_KEY);
-            }
+            checkLambda(ast, currentStatement);
+        }
+        else if (isResource(ast.getParent())) {
+            checkResourceVariable(ast);
         }
         else if (!inForHeader && isOnTheSameLine(currentStatement, lastStatementEnd,
                 forStatementEnd, lambdaStatementEnd)) {
             log(ast, MSG_KEY);
         }
     }
 
+    private void checkLambda(DetailAST ast, DetailAST currentStatement) {
+        int countOfSemiInCurrentLambda = countOfSemiInLambda.pop();
+        countOfSemiInCurrentLambda++;
+        countOfSemiInLambda.push(countOfSemiInCurrentLambda);
+        if (!inForHeader && countOfSemiInCurrentLambda > 1
+                && isOnTheSameLine(currentStatement,
+                lastStatementEnd, forStatementEnd,
+                lambdaStatementEnd)) {
+            log(ast, MSG_KEY);
+        }
+    }
+
+    private static boolean isResource(DetailAST ast) {
+        return ast != null
+            && (ast.getType() == TokenTypes.RESOURCES
+            || ast.getType() == TokenTypes.RESOURCE_SPECIFICATION);
```

Comment:
please add few more spaces before `||` , to make it +2 spaces after previous line `(`.
Order or execution should visual for code reader.

Reply:
done